### PR TITLE
kdePackages.kommit: init at 1.6.0

### PR DESCRIPTION
**Pull request by alex-fu27**
## Description of changes
Git GUI for KDE
-  homepage URL: https://apps.kde.org/kommit
-  source URL: https://invent.kde.org/sdk/kommit
-  license: GNU GPLv3
-  platforms: linux

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
(not applicable) Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
(not applicable) Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/misc/kommit/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  mkKdeDerivation,
+  fetchurl,
+  pkg-config,
+  libgit2,
+}:
+mkKdeDerivation rec {
+  pname = "kommit";
+  version = "1.6.0";
+
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/kde/misc/kommit/default.nix:12]: Do we have it already in the auto generated sources?

Comment (K900) [reply pkgs/kde/misc/kommit/default.nix:12]: No, it's not part of any of the large releases.

**Comment by alex-fu27** [comment]
This is a draft because there is a binary, `kommitmerge` wich tries to call `kommit`. That will fail if it's not on the `$PATH`. Now, do I have to patch something or can I pass something to `mkKdeDerivation` to add the output path to `$PATH` in the wrapper scripts?

**Comment by K900** [comment]
Can you hardcode the path instead?

**Comment by alex-fu27** [comment]
Yes. But if I run `kommitdiff`, `kommitmerge` or `kommit {diff,merge}`, then Kommit will segfault.

I only packaged this to evaluate Kommit and it turns out I won't actively use it. Would you like to have a package in nixpkgs that exists, but is unmaintained and does not completely work, or rather not have the package? In the latter case we should close the PR and hope that it is found by someone who is more interested in using Kommit.

**Comment by K900** [comment]
Not really, no. I'd rather wait for it to be officially incorporated into Gear or something.

**Pull request closed without merging**